pkg/install/update_clusterserviceprinicple.go

```diff
@@ -0,0 +1,164 @@
+		if err != nil {
+			return err
+		}
+
+		var config map[string]interface{}
+		err = json.Unmarshal([]byte(cpc.Data["config"]), &config)
+		if err != nil {
+			return err
+		}
+		tenantID, ok := config["tenantId"].(string)
+		if ok && tenantID == ssp.TenantID {
+			return nil
+		}
+		config["tenantID"] = ssp.TenantID
+		b, err := json.Marshal(config)
+		if err != nil {
+			return err
+		}
+		cpc.Data["config"] = string(b)
+		_, err = i.kubernetescli.CoreV1().ConfigMaps("openshift-config").Update(cpc)
+		return err
+	})
+}
+
+func (i *Installer) clusterSPObjectID(ctx context.Context) (string, error) {
+	var result string
+
+	spp := &i.doc.OpenShiftCluster.Properties.ServicePrincipalProfile
+
+	token, err := aad.GetToken(ctx, i.log, i.doc.OpenShiftCluster, azure.PublicCloud.GraphEndpoint)
+	if err != nil {
+		return "", err
+	}
+
+	spGraphAuthorizer := autorest.NewBearerAuthorizer(token)
+
+	applications := graphrbac.NewApplicationsClient(spp.TenantID, spGraphAuthorizer)
+
+	timeoutCtx, cancel := context.WithTimeout(ctx, 2*time.Minute)
+	defer cancel()
+	// NOTE: Do not override err with the error returned by wait.PollImmediateUntil.
+	// Doing this will not propagate the latest error to the user in case when wait exceeds the timeout
+	wait.PollImmediateUntil(10*time.Second, func() (bool, error) {
+		var res azgraphrbac.ServicePrincipalObjectResult
+		res, err = applications.GetServicePrincipalsIDByAppID(ctx, spp.ClientID)
+		if err != nil {
+			if strings.Contains(err.Error(), "Authorization_IdentityNotFound") {
+				i.log.Info(err)
+				return false, nil
+			}
+
+			return false, err
+		}
+
+		result = *res.Value
+		return true, nil
+	}, timeoutCtx.Done())
+
+	return result, err
+}
+
+func (i *Installer) updateRoleAssignments(ctx context.Context) error {
+	clusterSPObjectID, err := i.clusterSPObjectID(ctx)
+	if err != nil {
+		return nil
+	}
+	// TODO shouldn't this be the armAuthoriser?
+	roleassignments := authorization.NewRoleAssignmentsClient(i.env.SubscriptionID(), i.fpAuthorizer)
+
+	_, err = roleassignments.Create(ctx, "/subscriptions/"+i.env.SubscriptionID()+"/resourceGroups/"+i.env.ResourceGroup(), uuid.NewV4().String(), mgmtauthorization.RoleAssignmentCreateParameters{
+		RoleAssignmentProperties: &mgmtauthorization.RoleAssignmentProperties{
+			RoleDefinitionID: to.StringPtr("/subscriptions/" + i.env.SubscriptionID() + "/providers/Microsoft.Authorization/roleDefinitions/b24988ac-6180-42a0-ab88-20f7382dd24c"),
+			PrincipalID:      to.StringPtr(clusterSPObjectID),
+			PrincipalType:    mgmtauthorization.ServicePrincipal,
+		},
+	})
+	if detailedErr, ok := err.(autorest.DetailedError); ok {
+		if requestErr, ok := detailedErr.Original.(*azure.RequestError); ok &&
+			requestErr.ServiceError != nil &&
+			requestErr.ServiceError.Code == "RoleAssignmentExists" {
+			err = nil
+		}
+	}
+	/*	TODO, we also need to do this but there does not seem to be a create/update api
```

Comment:
Indeed.  Probably the easiest way to do this is to deploy a new ARM template with the updated details.  Actually that simplifies the error handling on the previous (roleassignments) step too.